aristanetworks/ansible-avd
# Feat(eos_designs): Add ability to set mlag port-channel id

## Change Summary

Gives users the ability to set the id of the port-channel used for the mlag peer link.

## Related Issue(s)

Fixes #1764

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Adding a field called `mlag_port_channel_id` which users can use in the node type defaults section or node group defaults section to set the mlag peer link port-channel id. If the key is not set, the default method of port-channel id generation is used.

Add a key called `mlag_port_channel_id` which can be set under node type defaults or node group defaults section.
```
< node_type_key >:
  mlag_port_channel_id: < 1-2000 > | Optional
  node_groups:
    <node_goup>:
      mlag_port_channel_id: < 1-2000 > | Optional
```

## How to test


With an inventory consisting of a pair of 7280s serving as l3leafs and a pair of 7308s serving as a custom node type called splines,  I added the key the splines defaults section only.  MLAG came up on the spline switches, which had the mlag port-channel id manually set, and MLAG also came up on the leaf switches which did NOT have the mlag port-channel id manually set. Both configurations matched the expected output.

## Checklist

### User Checklist


- N/A

### Repository Checklist



- [ x ] My code has been rebased from devel before I start
- [ x ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ x ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ x ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)
